qcodes/instrument/channel.py

```diff
@@ -529,3 +532,183 @@ def validate(self, value, context: str='') -> None:
+            **kwargs (dict): Keyword arguments needed to create the channels
+
+        Returns:
+            obj_list (list): List of instrument channels
+        """
+
+        obj_list = []
+        for new_kwargs in cls._discover_from_instrument(parent, **kwargs):
+            obj = cls(parent, existence=True, **new_kwargs)
+            obj_list.append(obj)
+
+        return obj_list
+
+    @classmethod
+    def _discover_from_instrument(
+            cls, parent: Instrument, **kwargs) ->List[dict]:
+        """
+        Discover channels on the instrument and return a list kwargs to create
+        these channels in memory
+
+        Args:
+            parent (Instrument): The instrument through which the instrument
+                channel is accessible
+            **kwargs (dict): Keyword arguments needed to discover the channels
+
+        Returns:
+              list: List of keyword arguments to initialize channels
+        """
+        raise NotImplementedError("Please subclass")
+
+    @classmethod
+    def get_new_instance_kwargs(cls, parent: Instrument, **kwargs)->dict:
+        """
+        When creating a new instance of this channel type,
+        this method will return kwargs to create this
+
+        Args:
+            parent: The instrument the new channel will belong to
+        """
+        raise NotImplementedError("Please subclass")
+
+    def __init__(
+            self,
+            parent: Union[Instrument, 'InstrumentChannel'],
+            name: str,
+            existence: bool=False,
+            channel_list: 'AutoLoadableChannelList'=None,
+            **kwargs
+    ) ->None:
+        """
+        Instantiate a channel object. Note that this is not the same as actually
+        creating the channel on the instrument. Parameters defined on this
+        channels will note be able to query/write to the instrument until it
+        has been created
+
+        Args:
+            parent (Instrument): The instrument through which the instrument
+                channel is accessible
+            name: channel name
+            existence (bool): True if the channel exists on the instrument
+            channel_list (N52xxChannelList): If the channel is deleted,
+                delete `self` from the channel list
+        """
+        super().__init__(parent, name=name, **kwargs)
+        self._exists_on_instrument = existence
+        self._channel_list = channel_list
+
+    def create(self) ->None:
+        """Create the channel on the instrument"""
+        self._create()
+        self._exists_on_instrument = True
+
+    def _create(self) ->None:
+        """
+        (SCPI) commands needed to create the instrument. Note that we
+        need to use `self.root_instrument.write` to send commands,
+        because self.write will cause _assert_existence to raise a
+        runtime error.
+        """
+        raise NotImplementedError("Please subclass")
+
+    def delete(self) ->None:
```

Comment:
should this do nothing if `self._exists_on_instrument` is False?

Reply:
If it is not on the instrument (yet), this method will probably raise an exception because `AssertExistance` will fail with a runtime error when we want to write or ask from the instrument. Never the less, as with `create`, we want to assert that the channel is present before deleting and raise a custom exception for this.